clients/include/blas2/testing_hpmv.hpp

```diff
@@ -78,50 +81,62 @@ void testing_hpmv_bad_arg(const Arguments& arg)
 
         if(arg.bad_arg_all)
         {
-            EXPECT_HIPBLAS_STATUS(
-                hipblasHpmvFn(handle, uplo, N, nullptr, dA, dx, incx, beta, dy, incy),
-                HIPBLAS_STATUS_INVALID_VALUE);
-            EXPECT_HIPBLAS_STATUS(
-                hipblasHpmvFn(handle, uplo, N, alpha, dA, dx, incx, nullptr, dy, incy),
-                HIPBLAS_STATUS_INVALID_VALUE);
+            DAPI_EXPECT(HIPBLAS_STATUS_INVALID_VALUE,
+                        hipblasHpmvFn,
+                        (handle, uplo, N, nullptr, dA, dx, incx, beta, dy, incy));
+
+            DAPI_EXPECT(HIPBLAS_STATUS_INVALID_VALUE,
+                        hipblasHpmvFn,
+                        (handle, uplo, N, alpha, dA, dx, incx, nullptr, dy, incy));
 
             if(pointer_mode == HIPBLAS_POINTER_MODE_HOST)
             {
                 // For device mode in rocBLAS we don't have checks for dA, dx, dy as we may be able to quick return
-                EXPECT_HIPBLAS_STATUS(
-                    hipblasHpmvFn(handle, uplo, N, alpha, nullptr, dx, incx, beta, dy, incy),
-                    HIPBLAS_STATUS_INVALID_VALUE);
-                EXPECT_HIPBLAS_STATUS(
-                    hipblasHpmvFn(handle, uplo, N, alpha, dA, nullptr, incx, beta, dy, incy),
-                    HIPBLAS_STATUS_INVALID_VALUE);
-                EXPECT_HIPBLAS_STATUS(
-                    hipblasHpmvFn(handle, uplo, N, alpha, dA, dx, incx, beta, nullptr, incy),
-                    HIPBLAS_STATUS_INVALID_VALUE);
+                DAPI_EXPECT(HIPBLAS_STATUS_INVALID_VALUE,
+                            hipblasHpmvFn,
+                            (handle, uplo, N, alpha, nullptr, dx, incx, beta, dy, incy));
+
+                DAPI_EXPECT(HIPBLAS_STATUS_INVALID_VALUE,
+                            hipblasHpmvFn,
+                            (handle, uplo, N, alpha, dA, nullptr, incx, beta, dy, incy));
+
+                DAPI_EXPECT(HIPBLAS_STATUS_INVALID_VALUE,
+                            hipblasHpmvFn,
+                            (handle, uplo, N, alpha, dA, dx, incx, beta, nullptr, incy));
             }
             // With alpha == 0 can have A and x nullptr
-            CHECK_HIPBLAS_ERROR(
-                hipblasHpmvFn(handle, uplo, N, zero, nullptr, nullptr, incx, beta, dy, incy));
+            DAPI_CHECK(hipblasHpmvFn,
+                       (handle, uplo, N, zero, nullptr, nullptr, incx, beta, dy, incy));
 
             // With alpha == 0 && beta == 1, all other ptrs can be nullptr
-            CHECK_HIPBLAS_ERROR(
-                hipblasHpmvFn(handle, uplo, N, zero, nullptr, nullptr, incx, one, nullptr, incy));
+            DAPI_CHECK(hipblasHpmvFn,
+                       (handle, uplo, N, zero, nullptr, nullptr, incx, one, nullptr, incy));
+        }
+
+        // Check 64-bit API with quick return
+        if(arg.api & c_API_64)
+        {
+            DAPI_CHECK(hipblasHpmvFn, (handle, uplo, N, zero, dA, dx, incx, one, dy, incy));
         }
```

Comment:
Why is this separate ?  Did you intend to set larger N ?

Reply:
This was intended to just call the 64 bit API and return success.

Reply:
> This was intended to just call the 64 bit API and return success.

All the DAPI_ calls call the 64bit API, so this case would be covered by the previous call above it already, no?

Reply:
Oh yeah you are right, I somehow missed it. In that case I could do a larger N and expect a invalid value status just for AMD backend.

Reply:
Well I would leave those to just be in rocBLAS only testing if you mean the thresholds for support.